optimum/intel/ipex/inference.py

```diff
@@ -20,7 +31,48 @@
     import intel_extension_for_pytorch as ipex
 
 
-class _ModelFallbackWrapper:
+def ordered_inputs(inputs, model):
+    ordered_inputs = {}
+    if hasattr(model, "forward"):
+        sig = inspect.signature(model.forward)
+    else:
+        sig = inspect.signature(model.call)
+
+    for param in sig.parameters:
+        param_regex = re.compile(rf"{param}(\.\d*)?")
+        to_insert = []
+        for name, dynamic_axes in inputs.items():
+            if re.match(param_regex, name):
+                to_insert.append((name, dynamic_axes))
+        # TODO: figure out a smart way of re-ordering potential nested structures.
+        to_insert = sorted(to_insert, key=lambda t: t[0])
+        for name, dynamic_axes in to_insert:
+            ordered_inputs[name] = dynamic_axes
+    return ordered_inputs
+
+
+def prepare_jit_inputs(model, task):
+    task = _TASK_ALIASES[task]
+    if model.config.use_cache:
+        task += "-with-past"
+    onnx_config_class = TasksManager.get_exporter_config_constructor(
+        exporter="onnx",
+        model=model,
+        task=task,
+    )
+    onnx_config = onnx_config_class(model.config)
+    dummy_inputs = onnx_config.generate_dummy_inputs(framework="pt")
+    dummy_inputs = ordered_inputs(dummy_inputs, model)
+
+    inputs = []
+    for k, v in dummy_inputs.items():
+        if v is not None and not isinstance(v, bool):
+            inputs.append(v)

```

Comment:
why is this needed ? if yes can this be integrated directly in `ordered_inputs` ?